packages/client/src/runtime/core/compositeProxy/createCompositeProxy.ts

```diff
@@ -104,6 +104,14 @@ export function createCompositeProxy<T extends object>(target: T, layers: Compos
             ...layer?.getPropertyDescriptor(prop),
           }
         }
+
+        if (prop === '$parent') {
```

Comment:
I think more appropriate way to do it would be to add optional way to define property descriptor for `addProperty` and set `enumerable: false` there. `createCompositeProxy` is meant to be very generic and know as little as possible about the contents of the objects it decorates.

Reply:
Definitely! this was just me trying things out and isolate what works. Thanks for the suggestion